File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0, right = 0, sum = 0;
        int minSubArraySize = Integer.MAX_VALUE;

        while (right < nums.length) {
            sum += nums[right++];

            while (sum >= target) {
                minSubArraySize = Math.min(minSubArraySize, right - left);
                sum -= nums[left++];
            }
        }

        return minSubArraySize > nums.length ? 0 : minSubArraySize;
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l = 0
        currSum = 0
        minWindow = float('inf')

        for r in range(len(nums)):
            currSum += nums[r]

            while currSum >= target:
                minWindow = min(minWindow, r - l + 1)
                currSum -= nums[l]
                l += 1
        
        return minWindow if minWindow != float('inf') else 0